sorting/go/quick-sort.go

```diff
@@ -0,0 +1,35 @@
+package main
```

Comment:
Hi @atin I suggest here to add also a comment to explain the quick sort and its complexity.

Reply:
Thanks @TawfikYasser , I've added some comments explaining quick sort.

Reply:
You’re welcome @atin 🙂